src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/AuditService.kt

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.service
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.entity.create.AuditAction
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.entity.create.AuditEntity
+import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.repository.AuditRepository
+import java.util.UUID
+
+@Service
+@Transactional
+class AuditService(private val auditRepository: AuditRepository) {
+  fun createAuditRecord(
```

Comment:
can the referral user name be removed from the signature since it's retrieved from the security context? 

Can we have convenience method that takes in a referral and works out all the fields from it? something like 

auditReferral(referral: ReferralEntity)
 it would work out all the fields AND whether the referralStatus was changing